chromium_src/chrome/browser/ssl/https_upgrades_navigation_throttle.cc

```diff
@@ -62,12 +63,34 @@ HttpsUpgradesNavigationThrottle::MaybeCreateThrottleFor(
   HostContentSettingsMap* map =
       HostContentSettingsMapFactory::GetForProfile(context);
 
-  bool https_first_mode_enabled =
+  PrefService* prefs = profile->GetPrefs();
+  security_interstitials::https_only_mode::HttpInterstitialState
+      interstitial_state;
+  interstitial_state.enabled_by_pref =
       (base::FeatureList::IsEnabled(features::kHttpsFirstModeV2) && prefs &&
        prefs->GetBoolean(prefs::kHttpsOnlyModeEnabled)) ||
       brave_shields::ShouldForceHttps(map, request_url);
+
+  StatefulSSLHostStateDelegate* state =
```

Comment:
I don't think these changes are correct. In particular StatefulSSLHostStateDelegate was left out of the original implementation and since we copied code (which we should almost never do) without any comments, it's not clear what the intention was, but now it's not clear at all what should be controlling this. This is a global pref for chromium, but it's a site setting for us so I think these other changes confuse what the actual setting is

Reply:
This whole thing is very confusing imo. If we want one behavior when IsHttpsByDefaultFeatureEnabled is enabled and the default chromium behavior when it's not, then we should always be deferring to the original chromium method, not copying the code here

Reply:
I can't recall why it was done this way, but I agree it doesn't seem ideal. I can enter an issue for this, but would probably need advice from someone like @arthuredelstein as far as how it should work.